text/0000-grid-inventory.md

```diff
@@ -0,0 +1,126 @@
+- Feature Name: (fill me in with a unique identifier, my_awesome_feature)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Hyperledger Grid Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes a generic and extensible framework for a Hyperledger Grid
+Inventory component, and is inspired by GS1’s EDI Inventory Report - Business
+Message Standard. GS1 is a widely used data standard in enterprises and, given
+that positioning and familiarity, Grid support feels natural. Additional
+implementations of Grid Inventory may derive from or extend this
+implementation.
+
+Grid Inventory leverages other extensible Hyperledger Grid components
+including: Grid Product, and Grid Location. Grid Inventory aims to provide
+current inventory quantity & status visibility for each Grid Product and Grid
+Location defined within the Grid Inventory smart contract between parties, as
+well as a summary of relevant inventory events that occurred within a
+time-range.  It includes transactions for generating summary inventory report
+information from external systems of record and recording the results in state.
+
+Grid Inventory also aims to serve as a foundation for future potential Grid
+capabilities that may require inventory related information, such as purchasing
+and logistics.
+
+# Motivation
+[motivation]: #motivation
+
+The Grid Inventory implementation is designed to enable shared visibility of
+summary inventory information for any defined Grid Product and Grid Location
+between participants.  Management of inventory levels is a common concept
+within supply chain solutions and is a key primary business activity related
+to all purchases of physical goods from suppliers, as well as all sales of
+physical goods to customers.  Management of inventory levels is typically
+coordinated very manually and very independently between parties across siloed
+systems and processes.  This siloed and manual coordination of inventory
+between parties can lead to substantial pain-points, including:
+inefficiencies, waste, and poor coordination of inventory levels across
+the supply chain.
+
+This Grid Inventory concept aims to solve these issues by offering a common
+industry solution for vertically integrating inventory information between
+parties existing systems of record.
+
+The Grid Inventory design addresses several use cases, including:
+
+- View Current Inventory Status Data: Viewing current summary inventory
+  quantities and inventory statuses between trading partners for defined Grid
+  Products and Grid Locations.
+
+- View Historical Inventory Activity Data: Viewing historical inventory
+  activity between trading partners for defined Grid Products and Grid
+  Locations.
```

Comment:
This should probably not allow smart software engineers to probe and consume information on inventory amounts such that information could be mined from a supplier such that those inventory and other attributes, like pricing, could be used against the supplier.

I read the `Motivation` section and jump to a `slickdeals` approach to when people should buy stuff. 

I assume there are guard rails and constraints and those would be great to elaborate here. An example would be that a customer is only setup for a certain amount of a supplier's product (X amount of product at Y interval with a Z price (or profit, internally)) and show the availability based on requested `promises` ... but not current and history data of inventory. Too vague, IMO.

Reply:
Hi Jason - thank you for this comment.  We should think about how the design can include appropriate security mechanisms to protect data being shared between consenting parties.